File: clients/customer/services/booking/customer.ts
import { bookingClient } from "./client";
import { SetLocationRequest } from "schema/booking/SetLocationRequest";
import { UpdateVIPRequest } from "schema/booking/UpdateVIPRequest";
import { GetCustomerResponse } from "schema/booking/GetCustomerResponse";


export async function updateCurrentLocation(req: SetLocationRequest) {
    return await bookingClient.post<SetLocationRequest, any>("/customer/set-location", req);
}

export async function updateVIP (req: UpdateVIPRequest) {
    return await bookingClient.post<UpdateVIPRequest, any>("/customer/vip", req);
}

export async function getCurrentCustomer () {
    return await bookingClient.get<GetCustomerResponse>("/customer");
}

File: services/booking/router/customer.go
package router

import (
	"database/sql"

	"github.com/gin-gonic/gin"
	"github.com/minhthao56/monorepo-taxi/services/booking/controller"
)

func NewRouterCustomer(r *gin.RouterGroup, conn *sql.DB) {
	controller := controller.NewCustomerController(conn)
	r.POST("/customer/set-location", controller.SetCurrentLocation)
	r.GET("/customers", controller.GetCustomers)
	r.POST("/customer/vip", controller.UpdateVIP)
	r.GET("/customer", controller.GetCurrentCustomer)
}
